site/app/controllers/HomePageController.php

```diff
@@ -32,13 +32,9 @@ public function __construct(Core $core) {
      * @Route("/home/courses", methods={"GET"})
      *
      * @param string|null $user_id
-     * @param bool|string $as_instructor
      * @return MultiResponse
      */
-    public function getCourses($user_id = null, $as_instructor = false) {
-        if ($as_instructor === 'true') {
-            $as_instructor = true;
-        }
+    public function getCourses($user_id = null) {
```

Comment:
When removing a function parameter, it's good practice to make sure to analyze why it was there in the first place. My preferred method to do this is to look at a combination of git blame / git history (which GitHub provides buttons for in the upper right when viewing a file). You should get led to the PR that added it, and then also how it's used ideally.

In this case, the `$as_instructor` flag was used here:
https://github.com/Submitty/Submitty/blob/47d4d9af9db157009b254c22e37a30be3f9cde6a/site/app/templates/CreateCourseForm.twig#L111-L113

where we only wanted to get courses the user was an instructor in to allow copying metadata from. Removal of the flag introduces a bug where that list may now contain courses the person is both an instructor or TA or student.

So yeah, let's revert this change.